source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -668,6 +670,11 @@ void ShaderGraph::finalize(ShaderGenerator& shadergen, const GenOptions& options
                         {
                             inputSocket = addInputSocket(interfaceName, input->type);
                             inputSocket->value = input->value;
+                            if (_renameTransformMap.find(inputSocket->name) != _renameTransformMap.end())
```

Comment:
You don't need the map at all -- and it's not very generic. If the input's variable is marked as "non-renamable", just copy over that input's variable name to the socket's variable name and then set the "non-renamable" flag. 

"noRename" is too vague a description for what this does. I think it would be better to add a flags variable and add flag to mark the variable as non-renamable. We can use the flag for the next work to mark inputs as to whether they can be optimized or not.